File: 1164 - Numero Perfeito/numero_perfeito.py
n = int(input())
for _ in range(n):
    x = int(input())
    soma = 0
    i = 1
    while i * i <= x:
        if x % i == 0:
            if i < x:
                soma += i
            j = x // i
            if j != i and j < x:
                soma += j
        i += 1
    if x > 0 and soma == x:
        print(f"{x} eh perfeito")
    else:
        print(f"{x} nao eh perfeito")

File: 1164 - Numero Perfeito/numero_perfeito.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    int n;
    cin >> n;
    while(n--){
        long long x;
        cin >> x;
        long long soma = 0;
        for(long long i = 1; i * i <= x; i++){
            if(x % i == 0){
                if(i < x) soma += i;
                long long j = x / i;
                if(j != i && j < x) soma += j;
            }
        }
        if(x > 0 && soma == x) cout << x << " eh perfeito" << endl;
        else cout << x << " nao eh perfeito" << endl;
    }
}
